pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuStateVec_helpers.hpp

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022-2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Adapted from JET: https://github.com/XanaduAI/jet.git
+
+/**
+ * @file cudaStateVec_helpers.hpp
+ */
+
+#pragma once
+#include <custatevec.h>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "cuStateVecError.hpp"
+
+namespace Pennylane::LightningGPU::Util {
+
+inline static auto pauliStringToEnum(const std::string &pauli_word)
+    -> std::vector<custatevecPauli_t> {
+    // Map string rep to Pauli enums
+    const std::unordered_map<std::string, custatevecPauli_t> pauli_map{
```

Comment:
May not matter, but I would be curious to know whether `unordered_map` are more performant than regular maps, or ordered maps for such small maps. Could a simple lookup in an array of pairs be optimal?

Reply:
Thanks @vincentmr ! Regarding the object construction per se, I would say the difference between `map` and `unordered_map` would be negligible (`unordered_map` would have slightly higher overhead due to the hashfunction..). However, per the following step of pauli_words to vector conversion, `unordered_map` would be better since the time complexity is `O(1)` ( `O(log(n))` for `map`). I would prefer to use `unordered_map` for performance purpose.

Reply:
Yes, my comment is about performance. For small sets, asymptotic complexity does not matter. Being an `O(1)` method is even a bad sign in my experience, but the implementation may adapt nicely to small set cases. I'd like us to study actual timings at some point, and hence I'm mentioning it.